workshops/micro_bit_flappy_bird/README.md

```diff
@@ -0,0 +1,363 @@
+
+```js
+let emptyObstacleY = 0
+let obstacles: game.LedSprite[] = []
+
+emptyObstacleY = randint(0, 4)
+```
+
+Using the `for` loop, iterate from 0 to 4. For every coordinate not equal to `emptyObstacleY` create and add obstacle sprites to the end of the obstacles array using the `push()` method.
+
+```js
+let emptyObstacleY = 0
+let obstacles: game.LedSprite[] = []
+
+emptyObstacleY = randint(0, 4)
+for (let index = 0; index <= 4; index++) {
+    if (index != emptyObstacleY) {
+        obstacles.push(game.createSprite(4, index))
+    }
+}
+```
+
+Now with every micro:bit restart you should see different autogenerated vertical obstacles.
+
+![random obstacles](https://cloud-70aevdanf.vercel.app/2random_obstacles.gif)
+
+Before continuing, make sure that obstacles are generated randomly and that the bird is moving up and down.
+
+## Step-4
+
+Let's make the obstacles move towards the bird. We have to make them move forever. The code inside `basic.forever()` keep running in the background. 
+
+Inside `forever()` access each obstacle using a `for` loop (iterate over the obstacles array) and decrease the obstacle `X` coordinate by `1`.
+
+```js
+let obstacles: game.LedSprite[] = []
+
+basic.forever(() => {
+    for (let obstacle of obstacles) {
+        obstacle.change(LedSpriteProperty.X, -1)
+    }
+    basic.pause(1000)
+})
+```
+
+Here, `basic.pause()` pauses the program for 1000 milliseconds(i.e., 1 second). Now obstacles should move towards the left every second.
+
+![moving obstacles](https://cloud-70aevdanf.vercel.app/3moving_onstacles.gif)
+
+## Step-5
+
+Now let's make obstacles disappear after reaching the leftmost corner. Iterate over all obstacles, delete the obstacle sprites where the `X` coordinate equals `0`, and remove them from the `obstacles` array. `get(property: LedSpriteProperty)` is used to know about the properties of a sprite.
+
+```js
+let obstacles: game.LedSprite[] = []
+
+basic.forever(() => {
+    while (obstacles.length > 0 && obstacles[0].get(LedSpriteProperty.X) == 0) {
+        obstacles.removeAt(0).delete()
+    }
+
+    for (let obstacle of obstacles) {
+        obstacle.change(LedSpriteProperty.X, -1)
+    }
+    basic.pause(1000)
+})
+```
+
+Now you see that when the obstacle reaches the bird it will just disappear.
+
+![disappearing obstacle](https://cloud-70aevdanf.vercel.app/4disappearing_obstacles.gif)
+
+## Step-6
+
+At the moment, our code generates just one vertical obstacle at a time. We need to put obstacle generation code into the `forever` loop so that it keeps generating more and more obstacles.
+
+```js
+let emptyObstacleY = 0
+let obstacles: game.LedSprite[] = []
+
+basic.forever(() => {
+    // all the previous code
+    
+    emptyObstacleY = randint(0, 4)
+    for (let index = 0; index <= 4; index++) {
```

Comment:
we don't know what's happening here and we might get confused. try to add a small code explanation here